walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -26,21 +26,41 @@
     //no attendees -- first test
     if (request.getAttendees() == null){
         availibility = Arrays.asList(TimeRange.WHOLE_DAY);
+       
+        //return the collection
+        return availibility;
     }
 
     //no options too long of a request (how do I access time range here? THis is an error) -- second test
     else if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
         availibility = Arrays.asList();
+        
+        //return the collection
+        return availibility;
     }
 
     //event splits day into two options (before and after) -- third test
     for (Event event : events){
```

Comment:
should there be an IF before this (if so, what should I be checking?) so that it can have its own return statement?

Reply:
You don't necessarily need an IF statement for each return statement.  What you have currently with the separate return statements for the special case above (also called "early return" or "short circuit" then the big calculation below is good.

Reply:
Ok just realized you have a loop here for test #3, and another loop below for test #4.  Ideally you'd want both loops combined into a single solution.  But if it helps to think through the solutions by doing them separately, feel free to put together separate solutions then combine/optimize later on.

What you can do if this is what you're going for, you can comment out `/*CODE_HERE*/` solutions that you know already works so they wouldn't affect solutions that you're still working on.